[ts5.8.2+node16] error TS2307: Cannot find module 'trusted-types/lib' or its corresponding type declarations

### Which package(s) are affected?

Lit Core (lit / lit-html / lit-element / reactive-element)

### Description

Upgrading to typescript@5.8.2, on a project that uses node16 module/moduleResolution, results in the following error:

```sh
> tsc

node_modules/lit-html/development/lit-html.d.ts:7:34 - error TS2307: Cannot find module 'trusted-types/lib' or its corresponding type declarations.

7 import type { TrustedHTML } from 'trusted-types/lib';
                                   ~~~~~~~~~~~~~~~~~~~


Found 1 error in node_modules/lit-html/development/lit-html.d.ts:7
```

### Reproduction

```
mkdir lit-type-issue
cd lit-type-issue
npm init -y
npm i typescript lit
npx tsc --init --skipLibCheck false --module node16 --moduleResolution node16
echo "import * as lit from 'lit'; console.log(lit);" > index.ts
npx tsc --noEmit
```

### Workaround

1. turn on skipLibCheck to avoid type-checking .d.ts files
2. avoid using the newest typescript version
3. avoid node16 moduleResolution

### Is this a regression?

Yes. This used to work, but now it doesn't.

### Affected versions

failing with typescript 5.8.2, passes with 5.7.3

### Browser/OS/Node environment

Browser: irrelevant
OS: Fedora 41
Node: 22.14.0
npm: 10.9.2

--- a/.changeset/wild-ways-deliver.md
+++ b/.changeset/wild-ways-deliver.md
@@ -1,0 +1,5 @@
+---
+'lit-html': patch
+---
+
+Import barrels explicitly for compatibility with modern Node resolution w/ ESM

--- a/.changeset/wild-ways-deliver.md
+++ b/.changeset/wild-ways-deliver.md
@@ -1,5 +1,6 @@
 ---
 'lit-html': patch
+'lit': patch
 ---
 
 Import barrels explicitly for compatibility with modern Node resolution w/ ESM

--- a/package.json
+++ b/package.json
@@ -160,6 +160,7 @@
         "./packages/localize-tools:test",
         "./packages/react:test",
         "./packages/tests:test",
+        "./packages/tests-typescript:test",
         "./packages/ts-transformers:test",
         "./packages/labs/analyzer:test",
         "./packages/labs/cli:test",

--- a/packages/labs/compiler/src/lib/template-transform.ts
+++ b/packages/labs/compiler/src/lib/template-transform.ts
@@ -4,7 +4,7 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-import type {TrustedHTML} from 'trusted-types/lib';
+import type {TrustedHTML} from 'trusted-types/lib/index.js';
 import ts from 'typescript';
 import {_$LH as litHtmlPrivate} from 'lit-html/private-ssr-support.js';
 import {parseFragment, serialize} from 'parse5';

--- a/packages/lit-html/src/lit-html.ts
+++ b/packages/lit-html/src/lit-html.ts
@@ -6,7 +6,7 @@
 
 // IMPORTANT: these imports must be type-only
 import type {Directive, DirectiveResult, PartInfo} from './directive.js';
-import type {TrustedHTML, TrustedTypesWindow} from 'trusted-types/lib';
+import type {TrustedHTML, TrustedTypesWindow} from 'trusted-types/lib/index.js';
 
 const DEV_MODE = true;
 const ENABLE_EXTRA_SECURITY_HOOKS = true;

--- a/packages/tests-typescript/package.json
+++ b/packages/tests-typescript/package.json
@@ -5,7 +5,7 @@
   "type": "module",
   "dependencies": {
     "typescript": "5.8.2",
-    "lit": "3.2.1"
+    "lit": "*"
   },
   "scripts": {
     "test": "wireit",
@@ -34,7 +34,7 @@
       ]
     },
     "test:bundler": {
-      "command": "npx tsc --module es2020 --moduleResolution bundler  --lib DOM --noEmit src/test-entry.ts",
+      "command": "npx tsc --module es2020 --moduleResolution bundler --lib DOM --noEmit src/test-entry.ts",
       "dependencies": [
         "../lit:build"
       ]

--- a/packages/tests-typescript/package.json
+++ b/packages/tests-typescript/package.json
@@ -1,0 +1,43 @@
+{
+  "name": "@lit-internal/tests-typescript",
+  "version": "0.0.1",
+  "private": true,
+  "type": "module",
+  "dependencies": {
+    "typescript": "5.8.2",
+    "lit": "3.2.1"
+  },
+  "scripts": {
+    "test": "wireit",
+    "test:node": "wireit",
+    "test:nodenext": "wireit",
+    "test:bundler": "wireit"
+  },
+  "wireit": {
+    "test": {
+      "dependencies": [
+        "test:node",
+        "test:nodenext",
+        "test:bundler"
+      ]
+    },
+    "test:node": {
+      "command": "npx tsc --module es2020 --moduleResolution node --lib DOM --noEmit src/test-entry.ts",
+      "dependencies": [
+        "../lit:build"
+      ]
+    },
+    "test:nodenext": {
+      "command": "npx tsc --module nodenext --moduleResolution nodenext --lib DOM --noEmit src/test-entry.ts",
+      "dependencies": [
+        "../lit:build"
+      ]
+    },
+    "test:bundler": {
+      "command": "npx tsc --module es2020 --moduleResolution bundler  --lib DOM --noEmit src/test-entry.ts",
+      "dependencies": [
+        "../lit:build"
+      ]
+    }
+  }
+}

--- a/packages/tests-typescript/src/test-entry.ts
+++ b/packages/tests-typescript/src/test-entry.ts
@@ -1,0 +1,2 @@
+import * as lit from 'lit';
+lit;

